.buildkite/scripts/commit.sh

```diff
@@ -16,35 +22,36 @@ if [ -z "${BUILDKITE}" ]; then
 	exit 1
 fi
 
-if [ "${BUILDKITE_BRANCH}" = "main" ]; then
-	echo "Skipping commit to main"
+# Commits only need to be done if:
+# (1) this is a PR build
+# (2) the quark submodule has changed compared to the base branch
+if [ -z "${BUILDKITE_PULL_REQUEST}" ]; then
+	echo "Skipping commit for non-PR build"
 	exit 0
-fi 
+fi
+
+if [ "$(git diff --submodule ${BUILDKITE_PULL_REQUEST_BASE_BRANCH} | wc -c)" = "0" ]; then
+	echo "Skipping commit; no change detected in quark submodule"
+	exit 0
+fi
 
 for ARCH in amd64 arm64; do
 	download libquark_big_${ARCH}.a .
 done
 
-# If there are no changes, don't commit
-if ! git diff --name-only HEAD | grep -q -E '\.a$'; then
-	echo "No changes detected"
-	exit 0
-fi
+for file in "${header_list[@]}"; do
+	cp src/${file} include/
+done
 
-# Don't commit if the last commit is from the bot, to avoid infinite loop of builds
-if test "$(git log -1 --pretty=format:'%ae')" = "${BOT_EMAIL}"; then
-	echo "last commit from bot"
-	exit 0
-fi
 
```

Comment:
This is not needed anymore? I'm guessing it's because the bot commit should never change `src/*` right?

Reply:
There's now a check at the pipeline upload step that will skip running the pipeline entirely if the last commit was from the bot: https://github.com/elastic/go-quark/blob/main/.buildkite/pipeline.yml#L23-L25